### DISCUSSION TITLE
Reduce latecy in read operation

### DISCUSSION THREAD

**Discussion by subin94** [apache/couchdb#3096] [Q&A]
When bulk inserts/updates are happening, the following read operations are slowing down. I believe it must be latency caused due to indexing.
From the documentation, https://docs.couchdb.org/en/latest/config/ioq.html, could see

> A server with a large number of views that do not need to be completely up-to-date may benefit from removing the bypass on view_update in order to optimize the latency for regular document read and write operations, and build the views during quieter periods.

- Does this mean configuring the value of `view_update` to false will improve read performance?
- Are there any costly trade-offs happening by configuring `view_update` to false ?
- And Also are there any other possible configuration changes to improve read performance?

Any help here will be greatly appreciated.
Thanks in Advance _/\\\_

**Comment by janl** [comment]
@subin94 which read operations are slow, and how much slower are they than before the bulk inserts? In addition, how many docs in total are you updating/inserting in your bulks?

**Reply by subin94** [reply]
> which read operations are slow

The read operation via `{db}/_find` .

>how much slower are they than before the bulk inserts?

Updates doesn't necessarily happen as a single bulk update, instead multiple bulk updates tends to happen in parallel. Could see at least a 3x delay when the _find request is made with other requests (with throughput around 300 RPM) than when made individually.

> how many docs in total are you updating/inserting in your bulks?

between 50-100 documents

Also there are 30+ indexes created in the database.

**Reply by janl** [reply]
ok, this sounds like the indexer needs to process all your inserts before it can respond to `_find`. This is normal behaviour.

If you are okay with returning results that don’t incorporate the latest inserts, you can use the `stale` option on your `_find` request: https://docs.couchdb.org/en/stable/api/database/find.html#db-find
